encord/project.py

```diff
@@ -340,7 +340,11 @@ def add_object(self, name: str, shape: ObjectShape) -> bool:
             OperationNotAllowed: If the operation is not allowed by the API key.
             ValueError: If invalid arguments are supplied in the function call
         """
-        return self._client.add_object(name, shape)
+        res = self._client.add_object(name, shape)
+        if res:
+            self._invalidate_project_instance()
```

Comment:
Why do we want to skip invalidation sometimes?

Reply:
Invalidation is not required if change request failed, e.g. if user has read-only access to the ontology, and request was rejected. 
But it's rather a corner case, so we might as well always invalidate, do not expect massive performance gains/hits from this checks

Reply:
To be honest, don't know, let me check, if exceptions are not handled, then yeah, this doesn't make sense

Reply:
Yeah, it seems we always throw, so no point in these checks, removed.
It also apparently means that in most of the cases there is no point in return value at all, but leaving them as is for now, don't want to change user-facing methods signatures without proper consideration

Reply:
@sergei-encord That is exactly the point of having exceptions,

you code the "happy" path, if something happens you just raise Exception to the top until it's catched,
this way we don't have and we should not care about "not happy" path this simplifies code a lot because you only write code as it would always work fine and you handle exceptions,

Could you imagine how complicated it would be to return None if exception is thrown?
Typing would think that None can be returned, user would think None can be returned because signature contains optional. But we would never return None because exception will stop the "not happy" path